src/Core/Context/CurrentContext.cs

```diff
@@ -298,16 +299,44 @@ public async Task<bool> AccessReports(Guid orgId)
                         && (o.Permissions?.AccessReports ?? false)) ?? false);
         }
 
-        public async Task<bool> ManageAllCollections(Guid orgId)
+        public async Task<bool> CreateNewCollections(Guid orgId)
+        {
+            return await OrganizationAdmin(orgId) || (Organizations?.Any(o => o.Id == orgId
+                        && (o.Permissions?.CreateNewCollections ?? false)) ?? false);
+        }
+
+        public async Task<bool> EditAnyCollection(Guid orgId)
+        {
+            return await OrganizationAdmin(orgId) || (Organizations?.Any(o => o.Id == orgId
+                        && (o.Permissions?.EditAnyCollection ?? false)) ?? false);
+        }
+
+        public async Task<bool> DeleteAnyCollection(Guid orgId)
         {
             return await OrganizationAdmin(orgId) || (Organizations?.Any(o => o.Id == orgId
-                        && (o.Permissions?.ManageAllCollections ?? false)) ?? false);
+                        && (o.Permissions?.DeleteAnyCollection ?? false)) ?? false);
+        }
+
+        public async Task<bool> ViewAllCollections(Guid orgId)
+        {
+            return await EditAnyCollection(orgId) || await DeleteAnyCollection(orgId);
         }
 
-        public async Task<bool> ManageAssignedCollections(Guid orgId)
+        public async Task<bool> EditAssignedCollections(Guid orgId)
         {
             return await OrganizationManager(orgId) || (Organizations?.Any(o => o.Id == orgId
-                        && (o.Permissions?.ManageAssignedCollections ?? false)) ?? false);
+                        && (o.Permissions?.EditAssignedCollections ?? false)) ?? false);
+        }
+
+        public async Task<bool> DeleteAssignedCollections(Guid orgId)
+        {
+            return await OrganizationManager(orgId) || (Organizations?.Any(o => o.Id == orgId
```

Comment:
I'm starting to think we should have a helper for this, something like:

```
HasOrgPermission(orgId, p => p.DeleteAssignedCollections);
```

Reply:
I thought about that too, but I think we'd still need to call the helper at every method. Ultimately it wouldn't reduce the boilerplate by much and that helper doesn't do anything for the manager vs admin checks that need to be done.

I think there's an unfortunate amount of boilerplate here, but it's ultimately for the best to limit the tribal knowledge to add a permission check and be explicit about what it's doing